packages/sit-onyx/src/components/OnyxDataGrid/features/filtering/filtering.ts

```diff
@@ -0,0 +1,132 @@
+import { createFeature } from "..";
+import { injectI18n } from "../../../../i18n";
+import { normalizedIncludes } from "../../../../utils/strings";
+import OnyxMiniSearch from "../../../OnyxMiniSearch/OnyxMiniSearch.vue";
+import type { OnyxMiniSearchProps } from "../../../OnyxMiniSearch/types";
+import OnyxSystemButton from "../../../OnyxSystemButton/OnyxSystemButton.vue";
+import type { DataGridEntry } from "../../types";
+import "./filtering.scss";
+import type { FilterOptions } from "./types";
+
+export const FILTERING_FEATURE = Symbol("Filtering");
+export type FilterState<TEntry> = Partial<Record<keyof TEntry, string | undefined>>;
+export const useFiltering = createFeature(
+  <TEntry extends DataGridEntry>(options?: FilterOptions<TEntry>) => {
+    const filters = ref({}) as Ref<FilterState<DataGridEntry>>;
+    const { t } = injectI18n();
+    const config = computed(() => toValue(options?.columns));
+
+    const isFilterEnabled = computed(() => (col: keyof TEntry) => {
+      return config.value?.[col]?.enabled !== false;
+    });
+
+    const filterData = (entries: Readonly<TEntry>[]) => {
+      return entries.filter((entry) =>
+        Object.entries(filters.value).every(
+          ([column, value]: [keyof TEntry, string | undefined]) => {
+            const columnOptions = config.value?.[column];
+            const filterOptions = { ...options?.filterConfig, ...columnOptions?.config };
+
+            if (value == null || value === "") return true;
+            const searchTerm = value.toString();
+            let entryValue = entry[column]?.toString() ?? "";
+
+            if (filterOptions.trimWhitespace) {
+              entryValue = entryValue.replace(/\s+/g, "");
+            }
+            if (filterOptions?.filterFunc) {
+              return filterOptions.filterFunc(entryValue, searchTerm);
+            }
+
+            if (filterOptions.exactMatch) {
+              return entryValue === searchTerm;
+            }
+
+            return normalizedIncludes(
+              entryValue,
+              searchTerm,
+              !filterOptions.caseSensitive,
+              filterOptions.searchFromStart,
+            );
+          },
+        ),
+      );
+    };
+
+    const clearFilter = (column: keyof DataGridEntry) => {
+      filters.value[column] = "";
+    };
+
+    // sync filters with user provided config
+    watchEffect(() => {
+      if (!config.value) return;
+      Object.entries(config.value).forEach(([column, columnOptions]) => {
+        if (columnOptions?.searchTerm) {
+          filters.value[column] = columnOptions.searchTerm;
+        }
+      });
+    });
+
+    const getMenuItem = (column: keyof DataGridEntry) => {
+      let inputValue = filters.value[column] || "";
+
+      const handleKeyDown = (e: KeyboardEvent) => {
+        if (e.key === "Enter") {
+          filters.value[column] = inputValue;
+        }
+      };
+
+      return h(OnyxMiniSearch, {
+        label: column.toString(),
```

Comment:
Please use a translated label like `Filter column {column}` so its more accessible and can easily be located in Playwright